admincontext.lua

```diff
@@ -0,0 +1,150 @@
+	end,
+
+	Action = function(self, entity)
+		self:MsgStart()
+			net.WriteEntity(entity)
+		self:MsgEnd()
+	end,
+
+	Receive = function(self, length, client)
+		if (client:IsAdmin()) then
+			local entity = net.ReadEntity()
+			client:NotifyLocalized(string.format(self.Format, entity:Nick(), entity:SteamID(), entity:Health(), entity:Armor()))
+		end
+	end
+})
+
+properties.Add("ixSetHealthProperty", {
+	MenuLabel = "#Health",
+	Order = 2,
+	MenuIcon = "icon16/heart.png",
+
+	Filter = function(self, entity, client)
+		return client:IsAdmin() and entity:IsPlayer()
+	end,
+
+	MenuOpen = function( self, option, ent, tr )
+
+		local submenu = option:AddSubMenu()
+		local target = IsValid( ent.AttachedEntity ) and ent.AttachedEntity or ent
+
+		for i = 100, 0, -25 do
+			local option = submenu:AddOption(i, function() self:SetHealth( ent, i ) end )
+		end
+
+	end,
+
+	Action = function(self, entity)
+		-- not used
+	end,
+
+	SetHealth = function(self, target, health)
+		self:MsgStart()
+			net.WriteEntity(target)
+			net.WriteUInt(health, 8)
+		self:MsgEnd()
+	end,
+
+	Receive = function(self, length, client)
+		if (client:IsAdmin()) then
+			local entity = net.ReadEntity()
+			local health = net.ReadUInt(8)
+
+			entity:SetHealth(health)
+			if (entity:Health() == 0) then entity:Kill() end
+		end
+	end
+})
+
+properties.Add("ixSetArmorProperty", {
+	MenuLabel = "#Armor",
+	Order = 3,
+	MenuIcon = "icon16/shield.png",
+
+	Filter = function(self, entity, client)
+		return client:IsAdmin() and entity:IsPlayer()
+	end,
+
+	MenuOpen = function( self, option, ent, tr )
+
+		local submenu = option:AddSubMenu()
+		local target = IsValid( ent.AttachedEntity ) and ent.AttachedEntity or ent
+
+		for i = 100, 0, -25 do
+			local option = submenu:AddOption(i, function() self:SetArmor( ent, i ) end )
+		end
+
+	end,
+
+	Action = function(self, entity)
+		-- not used
+	end,
+
+	SetArmor = function(self, target, armor)
+		self:MsgStart()
+			net.WriteEntity(target)
+			net.WriteUInt(armor, 8)
+		self:MsgEnd()
+	end,
+
+	Receive = function(self, length, client)
+		if (client:IsAdmin()) then
+			local entity = net.ReadEntity()
+			local armor = net.ReadUInt(8)
+
+			entity:SetArmor(armor)
+		end
+	end
+})
+
+properties.Add("ixSetDescriptionProperty", {
+	MenuLabel = "#Edit Description",
+	Order = 4,
+	MenuIcon = "icon16/book_edit.png",
+
+	Filter = function(self, entity, client)
+		return client:IsAdmin() and entity:IsPlayer()
+	end,
+
+	Action = function(self, entity)
+		self:MsgStart()
+			net.WriteEntity(entity)
+		self:MsgEnd()
+	end,
+
+	Receive = function(self, length, client)
+		if (client:IsAdmin()) then
+			local entity = net.ReadEntity()
+			client:RequestString("Set the character's description.", "New Description", function(text)
+				entity:GetCharacter():SetDescription(text)
+			end, entity:GetCharacter():GetDescription())
+		end
+	end
+
+})
+
+hook.Add("CanProperty", "bone_manipulate", function(ply, property, ent)
```

Comment:
This hook is unnecessary for the context of this plugin. Moreover, it's used incorrectly and will prevent **all** properties from being used